### ISSUE TITLE
Control Comments Broken Around Heredocs

### ISSUE THREAD

**Issue by phil4business** [puppetlabs/puppet-lint#18]
## Describe the Bug
puppet-lint does not respect control comments around heredocs. Similar to [rodjek/851](https://github.com/rodjek/puppet-lint/issues/851). puppet-lint warns of a missing ```lint:endignore``` even when present.

## Expected Behavior
puppet-lint should ignore checks disabled via control comments

## Steps to Reproduce
```puppet
class profile::example {
...
  # lint:ignore:variables_not_enclosed lint:ignore:variable_is_lowercase
  $pscommand =  @("RESPONSEHEADERCOMMAND"/$L)
    \$object = Get-WebConfiguration -PSPath ${$pspath} -Filter ${$filter}
    \$object.cacheControlMaxAge = '7.00:00:00'
    \$object.cacheControlMode = 'UseMaxAge'
    Set-WebConfiguration -PSPath ${$pspath} -Filter ${$filter} -InputObject \$object
  | RESPONSEHEADERCOMMAND
  # lint:endignore
...
}
```

Puppet-lint output: ```WARNING: lint:ignore:variables_not_enclosed comment on line 38 with no closing lint:endignore comment```. When multiple checks are provided on one line, the warning is always for the first check. The other checks do not produce this specific warning but are still checked (not ignored). E.x.:
```
PS C:\c\control-repo> puppet-lint --no-140chars-check .
WARNING: lint:ignore:variables_not_enclosed comment on line 38 with no closing lint:endignore comment
./site/profile/manifests/dig/ols/ext/d.pp - WARNING: variable not enclosed in {} on line 47 (check: variables_not_enclosed)
./site/profile/manifests/dig/ols/ext/d.pp - WARNING: variable not enclosed in {} on line 47 (check: variables_not_enclosed)
./site/profile/manifests/dig/ols/ext/d.pp - WARNING: variable not enclosed in {} on line 53 (check: variables_not_enclosed)
./site/profile/manifests/dig/ols/ext/d.pp - WARNING: variable contains an uppercase letter on line 47 (check: variable_is_lowercase)
```

If I put the control comments on separate lines then I just get the endignore error twice, even with two ```# lint:endignore``` comments.

The code generated in the heredoc is valid powershell and produces the expected output.

## Environment
 - Version 2.5.0
 - Platform Windows x64 via mingw32

**Comment by sanfrancrisko** [comment]
Thanks for reporting @phil4business - I can reproduce this issue on Windows, but not on macOS. I'm wondering if there's potentially an issue with Windows vs Unix line endings here.

I'll add this to our planning board
